File: solutions/csharp/scrabble-score/ScrabbleScore.cs
using System;
using System.Linq;

public static class ScrabbleScore
{
    public static int Score(string input) =>
        input.ToUpper().Aggregate(0, addLetters);

    private static int addLetters(int sum, char c) =>
        sum + letterValue(c);

    private static int letterValue(char letter) =>
        letter switch
        {
            'A' or 'E' or 'I' or 'O' or 'U' or
            'L' or 'N' or 'R' or 'S' or 'T' => 1,
            'D' or 'G' => 2,
            'B' or 'C' or 'M' or 'P' => 3,
            'F' or 'H' or 'V' or 'W' or 'Y' => 4,
            'K' => 5,
            'J' or 'X' => 8,
            'Q' or 'Z' => 10,
            _ => 0
        };
}

File: solutions/c/scrabble-score/scrabble_score.c
#include "scrabble_score.h"

#include <ctype.h>

unsigned int score(const char *input)
{
    static int LETTER_VALUE[] =
        {1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3,
         1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10};

    int sum = 0;
    for (char c = *input; c != '\0'; c = *++input)
    {
        sum += LETTER_VALUE[toupper(c) - 65];
    }
    return sum;
}
